src/parser.cc

```diff
@@ -2259,9 +2259,7 @@ void checkJointParentChildNames(const sdf::Root *_root, Errors &_errors)
       const std::string parentLocalName = sdf::SplitName(parentName).second;
 
       if (parentName != "world" && parentLocalName != "__model__" &&
-          !_model->LinkNameExists(parentName) &&
-          !_model->JointNameExists(parentName) &&
-          !_model->FrameNameExists(parentName))
```

Comment:
weird that we weren't checking if a joint parent matched a nested model name before now, while we were checking that for joint children

Reply:
Yeah, probably got missed when adding support for drill-down references to nested models.